Tribler/Main/tribler_main.py

```diff
@@ -145,30 +143,28 @@ def __init__(self, params, installdir, autoload_discovery=True,
         session = self.InitStage1(installdir, autoload_discovery=autoload_discovery,
                                   use_torrent_search=use_torrent_search, use_channel_search=use_channel_search)
 
-        self.splash = None
         try:
-            bm = self.gui_image_manager.getImage(u'splash.png')
-            self.splash = GaugeSplash(bm, "Loading...", 13)
-            self.splash.Show()
-
             self._logger.info('Client Starting Up.')
             self._logger.info("Tribler is using %s as working directory", self.installdir)
 
             # Stage 2: show the splash window and start the session
 
-            self.splash.tick('Starting API')
-            s = self.startAPI(session, self.splash.tick)
-
-            self.utility = Utility(self.installdir, s.get_state_dir())
+            self.utility = Utility(self.installdir, session.get_state_dir())
 
             if self.utility.read_config(u'saveas'):
                 DefaultDownloadStartupConfig.getInstance().set_dest_dir(self.utility.read_config(u'saveas'))
 
             self.utility.set_app(self)
-            self.utility.set_session(s)
+            self.utility.set_session(session)
```

Comment:
The `Session` object passed to `startAPI` is identical to the `Session` object returned by this method, making the return statement useless.